extensions/stackdriver/stackdriver.cc

```diff
@@ -645,6 +680,31 @@ void StackdriverRootContext::setConnectionState(
   tcp_request_queue_[id]->request_info->tcp_connection_state = state;
 }
 
+void StackdriverRootContext::evaluateExpressions(
+    std::unordered_map<std::string, std::string>& extra_labels) {
+  for (const auto& expression : expressions_) {
+    std::string value;
+    if (!evaluateExpression(expression.token, &value)) {
+      LOG_TRACE(absl::StrCat("Could not evaluate expression: ",
+                             expression.expression));
+      continue;
+    }
+    extra_labels[expression.tag] = value;
+  }
+}
+
+void StackdriverRootContext::cleanupExpressions() {
+  expressions_.clear();
+  for (auto const& item : tcp_request_queue_) {
```

Comment:
I am wondering under what circumstance would clean tcp request expression be useful?

Reply:
This is the question I had in mind.. what happens for a long standing tcp connection when configure is called in between? I thought for those, we need to have a cleanup and evaluate again...

/cc @kyessenov

Reply:
That will remove and recreate the root context I think.

Reply:
Actually, root context won't be recreated, it is just that onConfigure will be called once again. But looks like here the expr is cleaned but not refill in onConfigure, or did I miss anything? I actually think it is fine to leave existing TCP connection as is to avoid complexity.

Reply:
expressions will get created again from the proto config in onConfigure..

That's what I thought that, onConfigure will get called again. What about a scenario, wherein we want to debug long standing tcp connection and we add custom access log labels through our Telemetry API to see what is going on with this connection...

Can delete for now to keep TCP labels same and we can add this when we see the need..

Reply:
Yeah, it's an issue. Not sure how to handle it except making it fail safe in case some stale reports are referencing deleted expressions.

Reply:
Removed as we talked offline..